benchmarks/dynamo/huggingface.py

```diff
@@ -404,12 +395,31 @@ def load_model(
         else:
             config, model_cls = EXTRA_MODELS[model_name]
 
+        return model_cls, config
+
+    @download_retry_decorator
+    def _download_model(self, model_name):
+        model_cls, config = self._get_model_cls_and_config(model_name)
         if "auto" in model_cls.__module__:
             # Handle auto classes
-            model = model_cls.from_config(config).to(device, dtype=dtype)
+            model = model_cls.from_config(config)
         else:
-            model = model_cls(config).to(device, dtype=dtype)
+            model = model_cls(config)
+        return model
 
+    def load_model(
+        self,
+        device,
+        model_name,
+        batch_size=None,
+    ):
+        is_training = self.args.training
+        use_eval_mode = self.args.use_eval_mode
+        dtype = torch.float32
+        reset_rng_state()
+        model_cls, config = self._get_model_cls_and_config(model_name)
```

Comment:
Nit: This is called twice here and in `_download_model`, I guess you only need one here and pass them into `_download_model`?

Reply:
true, but in the case that we actually cache things it would be useful to make sure _download_model works with just the model name.